### DISCUSSION TITLE
Should we remove `idom.widgets.hotswap`?

### DISCUSSION THREAD

**Discussion by Archmonger** [idom-team/idom#865] [Ideas]
Ever since we've formalized support for conditionally rendered components, hotswap seems to have become a rather pointless utility.

This conversation is to discuss whether to deprecate/remove it.

**Comment by rmorshea** [comment]
The primary function isn't really to facilitate conditional rendering. Rather, it's to allow you to swap components from outside the normal rendering flow. With that said, it's uses are pretty niche. At the moment it has one usage in [IDOM's testing utilities](https://github.com/idom-team/idom/blob/1e3fac1e4227ac3d5ff225970c95e0e6acfef6ff/src/idom/testing/backend.py#L44). I'd say it would make sense to turn it into a private util in `idom.testing.backend`.
